populator-machinery/controller.go

```diff
@@ -108,13 +116,77 @@ type controller struct {
 	recorder             record.EventRecorder
 	referenceGrantLister referenceGrantv1beta1.ReferenceGrantLister
 	referenceGrantSynced cache.InformerSynced
+	populateFn           func(context.Context, *PopulatorParams) error
+	populateCompleteFn   func(context.Context, *PopulatorParams) (bool, error)
+	populateCleanupFn    func(context.Context, *PopulatorParams) error
+}
+
+type VolumePopulatorConfig struct {
+	MasterURL              string
+	Kubeconfig             string
+	ImageName              string
+	HttpEndpoint           string
+	MetricsPath            string
+	Namespace              string
+	Prefix                 string
+	Gk                     schema.GroupKind
+	Gvr                    schema.GroupVersionResource
+	PodConfig              *PodConfig
+	ProviderFunctionConfig *ProviderFunctionConfig
+}
+
+type PodConfig struct {
+	// MountPath is the path inside of the container at which the volume should be mounted
+	MountPath string
+	// DevicePath is the path inside of the container that the device will be mapped to
+	DevicePath string
+	// PopulatorArgs is the args for the populator pod
+	PopulatorArgs func(bool, *unstructured.Unstructured) ([]string, error)
+}
+
+type ProviderFunctionConfig struct {
+	// PopulateFn is the provider specific data population function
+	PopulateFn func(context.Context, *PopulatorParams) error
```

Comment:
Is there a reason why PopulatorParams needs to be a pointer? Will we ever pass in nil?

Reply:
Update to not pointer